### ISSUE TITLE
[rstudio-server] Can't overwrite file via uploading

### ISSUE THREAD

**Issue by vpanfilov** [rstudio/rstudio#7015]
### System details

    RStudio Edition :  Server
    RStudio Version :  1.3.959
    OS Version :  Ubuntu 20.04 in Docker (rocker/rstudio:4.0.0)
    R Version :  4.0.0

### Steps to reproduce the problem

1. Upload any file in RStudio Server via "Upload" button;
2. Try to upload the same file in the same place via "Upload" button. Click "Yes" in overwrite confirmation prompt dialog;
3. File is not uploaded with error "File exists"

### Describe the problem in detail
![изображение](https://user-images.githubusercontent.com/11993879/83408738-d1412080-a41b-11ea-8ffd-fea92c1da4e4.png)

### Describe the behavior you expected
File should be overwritten without errors.

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kfeinauer** [comment]
I am unable to reproduce with version 1.3.959 running on Ubuntu 16 - I don't suspect this is an issue on your particular operating system, but I won't rule that out.

Can you perform the action again and send me your RStudio Server logs as well as the exported request log? See details on how to do that here: https://support.rstudio.com/hc/en-us/articles/200488538-How-To-Export-a-Network-Request-Log-From-RStudio

**Comment by vpanfilov** [comment]
@kfeinauer I forgot to mention in the initial bug report that I run RStudio Server in Docker (rocker/rstudio:4.0.0). Futher analysis showed that this problem seems to be reproduced ONLY in mounted volumes inside docker container. But file permissions seems to be the same in mounted volume and in the home directory inside container:

```
Home directory inside container (no problems):
ruser@34c0e20aa8ca:~$ ls -la ~/cities.csv
-rw-r--r-- 1 ruser ruser 8402 Jun  2 14:08 /home/ruser/cities.csv

Mounted volume inside container (problem with reuploading):
ruser@34c0e20aa8ca:~$ ls -la ~/R/cities.csv 
-rw-r--r-- 1 ruser ruser 8402 Jun  2 13:56 /home/ruser/R/cities.csv
```

Sadly, I can't find the server logs inside container. Here are relevant request logs:

<details>
<summary>1) Initial upload request (sucess)</summary>

```json
Request ID: 501434454

== REQUEST ======
{
  "method": "complete_upload",
  "params": [
    {
      "filename": "cities.csv",
      "uploadedTempFile": "/home/ruser/.rstudio/uploaded-files/c76b-be67-771f-2c4e.bin",
      "targetDirectory": "/home/ruser/R"
    },
    true
  ],
  "clientId": "704c470c-814f-4d2d-9513-78cc5b831198",
  "clientVersion": "3a09be39fd51a8fafa8ae330007937d31924b395"
}

== RESPONSE ======
{
  "result": null,
  "ep": "false"
}
```

</details> 

<details>
<summary>2) Second upload request with overwrite (fail)</summary>

```json
Request ID: 722616819

== REQUEST ======
{
  "method": "complete_upload",
  "params": [
    {
      "filename": "cities.csv",
      "uploadedTempFile": "/home/ruser/.rstudio/uploaded-files/e84d-5fbe-d133-6d74.bin",
      "targetDirectory": "/home/ruser/R"
    },
    true
  ],
  "clientId": "704c470c-814f-4d2d-9513-78cc5b831198",
  "clientVersion": "3a09be39fd51a8fafa8ae330007937d31924b395"
}

== RESPONSE ======
{
  "error": {
    "code": 100,
    "message": "Error occurred while executing method",
    "error": {
      "code": 17,
      "category": "system",
      "message": "File exists",
      "location": "rstudio::core::Error rstudio::core::FilePath::copy(const rstudio::core::FilePath&) const src/cpp/shared_core/FilePath.cpp:604"
    }
  }
}
```

</details> 

<details>
<summary>3) Delete file request (success)</summary>

```json
Request ID: -1173363228

== REQUEST ======
{
  "method": "delete_files",
  "params": [
    [
      "~/R/cities.csv"
    ]
  ],
  "clientId": "704c470c-814f-4d2d-9513-78cc5b831198",
  "clientVersion": "3a09be39fd51a8fafa8ae330007937d31924b395"
}

== RESPONSE ======
{
  "result": null,
  "ep": "false"
}
```

</details> 

<details>
<summary>4) Third upload after file delete (success)</summary>

```json
Request ID: -537049414

== REQUEST ======
{
  "method": "complete_upload",
  "params": [
    {
      "filename": "cities.csv",
      "uploadedTempFile": "/home/ruser/.rstudio/uploaded-files/98f9-54dc-5749-34af.bin",
      "targetDirectory": "/home/ruser/R"
    },
    true
  ],
  "clientId": "704c470c-814f-4d2d-9513-78cc5b831198",
  "clientVersion": "3a09be39fd51a8fafa8ae330007937d31924b395"
}

== RESPONSE ======
{
  "result": null,
  "ep": "false"
}
```

</details> 

So I dont really know now, whether the problem is connected more with RStudio Server or with rocker-versioned project.
Should I prepare a reproducible example with current rocker RStudio image?

**Comment by kfeinauer** [comment]
You can enable RStudio Server logging to stderr by mounting the following file to `/etc/rstudio/logging.conf`:

```
[*]
logger-type=stderr
```

That should cause your server logs to show up in the Docker logs.

After performing step 2 and noting the upload failure, are you able to successfully run the following from the RStudio terminal to update the file contents?

```
echo "I am overwriting the file" > /home/ruser/R/cities.csv
```

Also, what are the file permissions on the entire directory tree on the uploaded file in step 2? It's possible the default umask for the file within the volume is causing issues here. Run the following from the terminal and send us the results:

```
namei -l /home/ruser/R/cities.csv
```

Thanks for your patience while we diagnose this!

**Comment by vpanfilov** [comment]
Sadly, I still can't see the logs with the proposed `/etc/rstudio/logging.conf` solution. Must be something to do with s6-init used in rocker images.

And yes, I can directly overwrite this file using console commands after upload failure:
```sh
ruser@e8bf0c0c7273:~$ head /home/ruser/R/cities.csv
"LatD", "LatM", "LatS", "NS", "LonD", "LonM", "LonS", "EW", "City", "State"
   41,    5,   59, "N",     80,   39,    0, "W", "Youngstown", OH
   42,   52,   48, "N",     97,   23,   23, "W", "Yankton", SD
   46,   35,   59, "N",    120,   30,   36, "W", "Yakima", WA
   42,   16,   12, "N",     71,   48,    0, "W", "Worcester", MA
   43,   37,   48, "N",     89,   46,   11, "W", "Wisconsin Dells", WI
   36,    5,   59, "N",     80,   15,    0, "W", "Winston-Salem", NC
   49,   52,   48, "N",     97,    9,    0, "W", "Winnipeg", MB
   39,   11,   23, "N",     78,    9,   36, "W", "Winchester", VA
   34,   14,   24, "N",     77,   55,   11, "W", "Wilmington", NC
ruser@e8bf0c0c7273:~$ echo "I am overwriting the file" > /home/ruser/R/cities.csv
ruser@e8bf0c0c7273:~$ cat /home/ruser/R/cities.csv
I am overwriting the file
```

Here is the output of `namei` command
```sh
$ namei -l /home/ruser/R/cities.csv
f: /home/ruser/R/cities.csv
drwxr-xr-x root  root  /
drwxr-xr-x root  root  home
drwxr-xr-x ruser ruser ruser
drwxr-xr-x ruser ruser R
-rw-r--r-- ruser ruser cities.csv
```

And the parent folder, inside of which this bug is not reproduced:
```sh
$ namei -l /home/ruser/cities.csv
f: /home/ruser/cities.csv
drwxr-xr-x root  root  /
drwxr-xr-x root  root  home
drwxr-xr-x ruser ruser ruser
-rw-r--r-- ruser ruser cities.csv
```

I prepared a minimal `docker-compose.yml` to reproduce this bug. It can be reproduced with `rocker/rstudio:4.0.0` image (RStudio 1.3.959 inside), but not with `rocker/rstudio:3.6.3` image (RStudio 1.2.5042 inside). `logging.conf` is also included but I can't get it working.

[upload-bug.zip](https://github.com/rstudio/rstudio/files/4722237/upload-bug.zip)

Place these files in empty folder, pull current images and start the container:
```sh
docker pull rocker/rstudio:4.0.0
docker-compose up
```
You can also test `rocker/rstudio:3.6.3` changing the **image** parameter in the **yml** file and restart with:
```sh
docker pull rocker/rstudio:3.6.3
docker-compose up
```

**Comment by vpanfilov** [comment]
The same problem is present with `Move...` command, if I am trying to move the file with the same name into **/home/ruser/R** directory. But, strangely, `Copy to...` command seems to be working fine in this case.

**Comment by homerhanumat** [comment]
I get this issue with R Studio Pro 1.3.959-1, "Middlemist Red" (feab7a49, 2020-05-18) for CentOS 6.

I am allowed to over-write a file in my Home directory, but not in the directory `/com` to which I have rwx access by virtue of membership in a group.

I attempted to download the Request Log as per [ https://support.rstudio.com/hc/en-us/articles/200488538-How-To-Export-a-Network-Request-Log-From-RStudio]( https://support.rstudio.com/hc/en-us/articles/200488538-How-To-Export-a-Network-Request-Log-From-RStudio) but no download occurs.

Can I get the request log some other way (e.g., ssh)?

**Comment by kfeinauer** [comment]
@homerhanumat We believe we have identified and fixed the cause of this issue, and it should be fixed in version `1.3.1009`, which you can try in a preview build: https://rstudio.com/products/rstudio/download/preview/
